nunaliit2-multimedia/src/main/java/ca/carleton/gcrc/olkit/multimedia/ffmpeg/FFprobeProcessor.java

```diff
@@ -0,0 +1,54 @@
+package ca.carleton.gcrc.olkit.multimedia.ffmpeg;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import java.io.BufferedReader;
+import org.json.JSONObject;
+import org.json.JSONArray;
+import java.util.stream.Collectors;
+
+import java.util.List;
+import java.util.ArrayList;
+
+public class FFprobeProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(FFprobeProcessor.class);
+
+    static public String ffprobeCommand = "ffprobe -v quiet -print_format json -show_streams %s";
+
+    private FFprobeProcessor() {
+    }
+
+    public static List<FileStream> getFileStreams(File file) {
+        String cmd = String.format(ffprobeCommand, file.getPath());
+        Process process = null;
+        try {
+            process = Runtime.getRuntime().exec(cmd);
+            BufferedReader br = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String output = br.lines().collect(Collectors.joining("\n"));
+            JSONObject json = new JSONObject(output);
+            JSONArray streams = json.getJSONArray("streams");
+
+            List<FileStream> fileStreams = new ArrayList<FileStream>();
+            for (int i = 0; i < streams.length(); i++) {

```

Comment:
Could consider using streams, would make it cleaner IMO but not a big deal if you prefer this appraoch.

Reply:
https://mkyong.com/java8/java-8-streams-map-examples/